lib/libmage6.py

```diff
@@ -0,0 +1,155 @@
+        except requests.ConnectionError, e:
+            self._session = None
+            msg = "Failed to connect to MAGE: %s" % e
+            self.logger.fatal(msg)
+            self.logger.exception(e)
+            raise LibMageException(msg)
+
+        if request.status_code != 200:
+            self._session = None
+            msg = "Cannot log in Mage. Response was %s: %s" % (request.status_code, request.reason)
+            self.logger.fatal(msg)
+            raise LibMageException(msg)
+
+
+    def logout(self):
+        """Be a good citizen, close session"""
+        #TODO: call mage logout URL
+        pass
+
+
+    def run_query(self, query, unique=False):
+        """Execute a query on mage. Return results as json like dict
+        @:param unique: if True, assume that query return only one. It raises if stricly less or more that 1 response it returned. Default is False
+        @:return list of responses. If uniq=True, returns only one response object instead of a list of response.. """
+        base_query = "ref/mql/json/"
+        url = urljoin(self.base_url, base_query)
+        url = urljoin(url, query)
+        if self._session is None:
+            msg = "You must login before trying to run a query"
+            self.logger.fatal(msg)
+            raise LibMageException(msg)
+        self.logger.debug("Running mage query: %s" % url)
+        request = self._session.get(url)
+        if request.status_code != 200:
+            msg = "Cannot query Mage. Response was %s: %s" % (request.status_code, request.reason)
+            self.logger.critical(msg)
+            raise LibMageException(msg)
+        self.logger.debug("Query answer: %s" % request.json())
+        if unique:
+            response = request.json()
+            if len(response) == 1:
+                return response[0]
+            else:
+                msg = "Unique query did not return exactly one result. Got %s" % len(response)
+                self.logger.critical(msg)
+                raise LibMageException(msg)
+        else:
+            response = request.json()
+        return response
+
+
+    def mage_get_delivery_id(self):
+        raise NotImplementedError()
+
+
+    def mage_get_delivery_content(self):
+        raise NotImplementedError()
+
+
+    def mage_get_installable_item_detail(self):
+        raise NotImplementedError()
+
+
+    def mage_get_install_methods(self):
+        raise NotImplementedError()
+
+
+    def mage_get_compatible_installable_items(self):
+        raise NotImplementedError()
+
+
+    def mage_test_ii_dependencies(self):
+        raise NotImplementedError()
+
+
+    def mage_register_install(self):
+        raise NotImplementedError()
+
+
+    def mage_register_backup_ci(self):
+        raise NotImplementedError()
+
+
+    def mage_archive_set(self):
+        raise NotImplementedError()
+
+
+    def mage_latest_backup_age(self):
+        raise NotImplementedError()
+
+
+class LibMageException(Exception):
```

Comment:
That would create a (hard) dependency to MAGE core. For now, the libmage6.py file can be used as a standalone file. It would be nice to keep this property.
But I can create specific client exception that are consistent with the two you propose. Something like LibMageInternalError and LibMageCallerError. Ok with that ?